src/components/UserPage/UserPage.js

```diff
@@ -1,10 +1,135 @@
+})
+
+const UserPageFunc = ({
+  dispatch,
+  history,
+  user,
+  friends,
+  decks,
+}) => {
+  useEffect(() => {
+    dispatch({ type: 'FETCH_USER' });
+    dispatch({ type: 'GET_DECK' });
+    dispatch({ type: 'GET_FRIENDS' });
+  }, [dispatch])
+
+  const createDeck = () => {
+    const deck = prompt("Please enter a deck name:", "My Awesome Deck");
+    if (deck === null || deck === "") {
+      alert("Canceled");
+    } else {
+      this.props.dispatch({
+        type: 'CREATE_DECK',
+        payload: {
+          userid: user.id,
+          deckname: deck,
+          ispublic: false,
+          description: "",
+          decklist: '',
+          featured_card: '',
+          upvotes: 0,
+          comments: '',
+        },
+      });
+    };
+  };
+
+  const deleteDeck = (deck) => {
+    const response = window.confirm(`Are you sure you want to permanenently delete this deck?`)
+    if (response){
+      this.props.dispatch({
+        type: 'DELETE_DECK',
+        payload: deck
+      })
+    }
+  };
+
+  return (
+    <div id='userMainDiv'>
+      <img id='profilePic' height='130px' width='130px' src={user.img_url} />
+      <div id='profileInfo'>
+        <h1 id='welcome'>
+          Welcome, {user.username}!
+        </h1>
+        <button onClick={() => history.push('/edituser')}>
+          Edit Profile
+        </button>
+      </div>
+      <hr />
+      <div id='mainDiv'>
+        <div id='userDiv'>
+          <h1>
+            {user.username}'s Decks
+          </h1>
+          <br/>
+          <button onClick={createDeck}>
+            Create New Deck
+          </button>
+          <hr/>
+          <div id='userDeckScroll'>
+            {decks.map((deck) =>
+              <div id='deckOptions' key={deck.id}>
+                <h4 id='deckName' onClick={() => history.push(`/viewdeck/${deck.id}`)}>
+                  {deck.deckname}
+                </h4>
+                <h5 id='upvotes'>
+                  Upvotes: {deck.upvotes}
+                </h5>
+                <button onClick={() => history.push(`/editdeck/${deck.id}`)}>
+                  EDIT
+                </button>
+                <button onClick={() => deleteDeck(deck.id)}>
+                  DELETE
+                </button>
+                <br/>
+              </div>
+            )}
+          </div>
+        </div>
+        <div id='userDiv'>
+          <h1>
+            {user.username}'s Friends
+          </h1>
+          <br/>
+          <button onClick={() => console.log('this.viewUsers?')}>
+            View/Search Users
+          </button>
+          <hr/>
+          <div id='friendsDeckScroll'>
+            {friends
+              .filter(f => f.id !== user.id)
+              .map((friend) =>
+                <div id='friendOptions' key={friend.id}>
+                  <h4 id='deckName'>
+                    {friend.username}
+                  </h4>
+                </div>
+              )
+            }
+          </div>
+        </div>
```

Comment:
Since decks and friends are not connected they could and maybe should be separate components. That way we can separate the concerns of each piece of the page for which part of the application state they care about and need to know about. As it stands now if we add a deck, friends has to re-render, because it is connected to the deck state in redux.